homeassistant/components/filter/sensor.py

```diff
@@ -423,6 +423,9 @@ def __init__(
 
     def _filter_state(self, new_state):
         """Implement the range filter."""
+        if not isinstance(new_state.state, Number):
```

Comment:
This is a weird fix. I think that the correct fix would be for `FilterState` to not even be instantiated and the value skipped?

Reply:
What happens is that some filters work with non Numbers (2 throttle filter's)

Reply:
Then maybe add a boolean to filters to specify that ? it seems weird to just copy paste a solution to every filter and every future filter.

Reply:
I can do that, will update later tonight